File: 0001-two-sum/solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer  ,Integer> map = new HashMap<Integer, Integer>();
        int pre = 0 ;
        int nex = 0;
        for(int i  =0  ; i<nums.length ; i++){
            //System.out.println(i+"  "+(target-nums[i]));
            if( map.containsKey(nums[i])) {
                
                System.out.println(map.get(target-nums[i])) ;                                   
                                                 pre = i;
                                                  nex = map.get(nums[i]);
                                                 break;
                                                 }
            else map.put((target-nums[i] ), i);
            
            //for(int j : map.keySet()) System.out.println(j+"    "+map.get(j));
            //System.out.println(" ////// ");
        }
        //System.out.println(pre+"  "+nex);
        return new int[]{pre ,nex};
    }
}

File: 0001-two-sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for index,value in enumerate(nums):
            # print(f" index {index} with value {value} and remaining list is {nums[index+1:]} and target-value is {target-value}")
            nextHalf =nums[index+1:]
            if target-value in nextHalf:
                return [index, 1+index+nextHalf.index(target-value)] 

        return [0,0]
